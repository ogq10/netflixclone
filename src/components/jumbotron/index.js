import React from 'react';
import {
  Container,
  Item,
  Inner,
  Title,
  Left,
  Right,
  SubTitle,
  Image,
} from './styles/jumbotron';

export default function Jumbotron({
  children,
  direction = 'row',
  ...restProps
}) {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Left = function JumbotronContainer({ children, ...restProps }) {
  return <Left {...restProps}>{children}</Left>;
};

Jumbotron.Title = function JumbotronContainer({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumbotronContainer({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Right = function JumbotronContainer({ children, ...restProps }) {
  return <Right {...restProps}>{children}</Right>;
};

Jumbotron.Image = function JumbotronContainer({ ...restProps }) {
  return <Image {...restProps} />;
};
