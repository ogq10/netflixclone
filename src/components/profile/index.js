import React from "react";
import { Container, Title, Item, List, Name, Picture } from "./styles/profile";

export default function Profile({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Profile.Title = function ProfileTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Profile.User = function ProfileUser({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

Profile.List = function ProfileList({ children, ...restProps }) {
  return <List {...restProps}>{children}</List>;
};

Profile.Name = function ProfileName({ children, ...restProps }) {
  return <Name {...restProps}>{children}</Name>;
};

Profile.Picture = function ProfilePicture({ src, ...restProps }) {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.png`}
    />
  );
};
