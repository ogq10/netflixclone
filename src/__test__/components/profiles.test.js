import React from "react";
import { render } from "@testing-library/react";
import { Profile } from "../../components";

describe("<Profile />", () => {
  it("renders the <Profile /> with populated data", () => {
    const { container, getByText, getByTestId } = render(
      <Profile>
        <Profile.Title>Who's watching?</Profile.Title>
        <Profile.List>
          <Profile.User onClick={() => {}}>
            <Profile.Picture
              src="/images/karl.png"
              data-testid="profile-picture"
            />
            <Profile.Name>Omar</Profile.Name>
          </Profile.User>
        </Profile.List>
      </Profile>
    );

    expect(getByText("Who's watching?"));
    expect(getByTestId("profile-picture")).toBeTruthy();
    expect(getByText("Omar"));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Profile /> with populated data but misc profile image", () => {
    const { container, getByText, getByTestId } = render(
      <Profile>
        <Profile.Title>Who's watching?</Profile.Title>
        <Profile.List>
          <Profile.User onClick={() => {}}>
            <Profile.Picture data-testid="profile-picture-misc" />
            <Profile.Name>Omar</Profile.Name>
          </Profile.User>
        </Profile.List>
      </Profile>
    );

    expect(getByText("Who's watching?"));
    expect(getByTestId("profile-picture-misc")).toBeTruthy();
    expect(getByText("Omar"));
    expect(container.firstChild).toMatchSnapshot();
  });
});
