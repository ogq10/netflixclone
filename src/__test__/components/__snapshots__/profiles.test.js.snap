// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders the <Profile /> with populated data 1`] = `
<div
  class="profile__Container-sc-11t8kut-0 guPoni"
>
  <h1
    class="profile__Title-sc-11t8kut-1 iYucNC"
  >
    Who's watching?
  </h1>
  <ul
    class="profile__List-sc-11t8kut-2 epvZZl"
  >
    <li
      class="profile__Item-sc-11t8kut-5 cukbhD"
    >
      <img
        class="profile__Picture-sc-11t8kut-4 keGOOu"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profile__Name-sc-11t8kut-3 gqtDGN"
      >
        Omar
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profile /> renders the <Profile /> with populated data but misc profile image 1`] = `
<div
  class="profile__Container-sc-11t8kut-0 guPoni"
>
  <h1
    class="profile__Title-sc-11t8kut-1 iYucNC"
  >
    Who's watching?
  </h1>
  <ul
    class="profile__List-sc-11t8kut-2 epvZZl"
  >
    <li
      class="profile__Item-sc-11t8kut-5 cukbhD"
    >
      <img
        class="profile__Picture-sc-11t8kut-4 keGOOu"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.png"
      />
      <p
        class="profile__Name-sc-11t8kut-3 gqtDGN"
      >
        Omar
      </p>
    </li>
  </ul>
</div>
`;
